lib/datasource/repology/index.ts

```diff
@@ -104,138 +43,188 @@ function findPackageInResponse(
+        // 403 is handled later because in this case we are trying the API
+        return [];
+      }
 
-async function getCachedPackage(
-  registryUrl: string,
-  repoName: string,
-  pkgName: string
-): Promise<RepologyPackage[]> {
-  // Fetch previous result from cache if available
-  const cacheKey = joinUrlParts(registryUrl, repoName, pkgName);
-  const cachedResult = await packageCache.get<RepologyPackage[]>(
-    cacheNamespace,
-    cacheKey
-  );
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
+      throw err;
+    }
   }
 
-  // Attempt a package lookup and return if found non empty list
-  const pkg = await queryPackage(registryUrl, repoName, pkgName);
-  if (pkg && pkg.length > 0) {
-    await packageCache.set(cacheNamespace, cacheKey, pkg, cacheMinutes);
-    return pkg;
-  }
+  private async queryPackagesViaResolver(
+    registryUrl: string,
+    repoName: string,
+    packageName: string,
+    packageType: RepologyPackageType
+  ): Promise<RepologyPackage[]> {
+    const query = getQueryString({
+      repo: repoName,
+      name_type: packageType,
+      target_page: 'api_v1_project',
+      noautoresolve: 'on',
+      name: packageName,
+    });
+
+    // Retrieve list of packages by looking up Repology project
+    const packages = await this.queryPackages(
+      joinUrlParts(registryUrl, `tools/project-by?${query}`)
+    );
 
-  // No package was found on Repology
-  return null;
-}
+    return packages;
+  }
 
-export async function getReleases({
-  lookupName,
-  registryUrl,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  // Ensure lookup name contains both repository and package
-  const [repoName, pkgName] = lookupName.split('/', 2);
-  if (!repoName || !pkgName) {
-    throw new ExternalHostError(
-      new Error(
-        'Repology lookup name must contain repository and package separated by slash (<repo>/<pkg>)'
-      )
+  private async queryPackagesViaAPI(
+    registryUrl: string,
+    packageName: string
+  ): Promise<RepologyPackage[]> {
+    // Directly query the package via the API. This will only work if `packageName` has the
+    // same name as the repology project
+    const packages = await this.queryPackages(
+      joinUrlParts(registryUrl, `api/v1/project`, packageName)
     );
+
+    return packages;
   }
 
-  logger.trace(`repology.getReleases(${repoName}, ${pkgName})`);
-  try {
-    // Attempt to retrieve (cached) package information from Repology
-    const pkg = await getCachedPackage(registryUrl, repoName, pkgName);
-    if (!pkg) {
-      return null;
+  @cache({
+    ttlMinutes: 60,
+    namespace: `datasource-${RepologyDatasource.id}-list`,
+    key: (registryUrl: string, repoName: string, pkgName: string) =>
+      joinUrlParts(registryUrl, repoName, pkgName),
+  })
+  async queryPackage(
+    registryUrl: string,
+    repoName: string,
+    pkgName: string
+  ): Promise<RepologyPackage[] | undefined> {
```

Comment:
`undefined` is used instead of `null`, because decorator doesn't cache `undefined` values

Reply:
Why we don't want to cache?